interfaces/console/src/components/SiteInfos/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import { Grid, Paper, Stack, Typography, Skeleton } from '@mui/material';
+import { SiteDto } from '@/client/graphql/generated';
+import dynamic from 'next/dynamic';
+import { useAppContext } from '@/context';
+
+const SiteMapComponent = dynamic(() => import('../SiteMapComponent'), {
+  ssr: false,
+});
+
+interface SiteInfoProps {
+  selectedSite: SiteDto;
+}
+
+const SiteInfo: React.FC<SiteInfoProps> = ({ selectedSite }) => {
+  const { setSnackbarMessage, setSelectedDefaultSite } = useAppContext();
+  const [address, setAddress] = useState('');
+
+  const fetchAddress = async (lat: number, lng: number) => {
+    return await fetch(
+      `https://nominatim.openstreetmap.org/reverse?format=json&lat=${lat || 37.7749}&lon=${lng || -122.4194}`,
+      {
+        cache: 'force-cache',
+      },
+    )
+      .then((res) => res.json())
+      .then((data) => data.display_name)
+      .catch(() => 'Location not found');
+  };
```

Comment:
Seems like we are using this method multiple places, Can we common it out. Create a `repo` directory under `src` and move it to there.